DEB Unit Tests

jag2k2 opened the pull request:
- [x] This contribution adheres to [contributing.md](docs/contributing.md).

### What does this Pull Request accomplish?

This pull request adds unit tests for the Deterministic Error Building library

### Why should this Pull Request be merged?


### What testing has been done?

Commits on this pull request have been tested using the JKI VI Tester

jag2k2 committed:
Added Error Cluster creation unit tests

jag2k2 committed:
Adding Parameter Setting Unit Tests

jag2k2 committed:
Adding Write to Source Test.  Will add other datatypes to this VI

jag2k2 committed:
Merge branch 'v1.0.0' into feature/deb_unit_test

jag2k2 committed:
Updated to test against string constants

jag2k2 committed:
Removing test until we decide what to do with this VI

jag2k2 committed:
Finished test for Writes to Source

jag2k2 committed:
Added test for string writes to buffer

jag2k2 committed:
Found bug in Set Error Description and am removing it since its functionality is mostly redundant

jag2k2 committed:
Refactored to test against constants

jag2k2 committed:
Cleaned up block diagrams

jag2k2 committed:
updating vipb to reflect removal of Set Error Description.vi

csjall commented:
- [x] `Creation\Creation.ctl`, `Creation\setUp.vi`, `Creation\tearDown.vi`: Remove "None" parameter from class. It is created by default and is not part of your tests.
![image](https://user-images.githubusercontent.com/4081815/63720144-28262380-c814-11e9-9f47-8cc474d2552b.png)
![image](https://user-images.githubusercontent.com/4081815/63720099-0a58be80-c814-11e9-83ab-2b5eaaa6866e.png)
![image](https://user-images.githubusercontent.com/4081815/63720203-4b50d300-c814-11e9-907d-f8e036fa24ce.png)
- [x] Remove `temp_VI_UnderTest.vi` from test classes. This is an example that is not needed in your test class. You will have to remove the reference in `testExample.vit`.
![image](https://user-images.githubusercontent.com/4081815/63720921-d41c3e80-c815-11e9-9855-da6f38840070.png)
- [x] `Creation\testCreation256.vi`, `Creation\testCreation512.vi`, `Creation\testCreation1024.vi`: Wire error in to passIfEqual.
![image](https://user-images.githubusercontent.com/4081815/63721040-0f1e7200-c816-11e9-8405-82f00c71d61b.png)
- [x] `test Writes to Source.vi`: This violates my "your block diagram should fit on a single screen" rule of thumb. Consider using subVIs or splitting into separate tests. This improves readability by allowing tech leads and developers to survey the test without scrolling 3 screens to the right.
- [x] `test Writes to Source.vi`: Remove stray constant on block diagram.
![image](https://user-images.githubusercontent.com/4081815/63722077-15155280-c818-11e9-8a07-75b132e34836.png)
- [ ] You have positive tests which confirm that valid parameters work as expected. Consider adding some negative tests which confirm that invalid parameters (i.e. strings that exceed static buffer length) return an error.

jag2k2 committed:
Removed "None" parameter from class and its methods

jag2k2 committed:
Removing temp_VI_UnderTest.vi from both classes and vi templates

jag2k2 committed:
Wired "error in" of each of these VIs to the error in terminal of passIfEqual

jag2k2 committed:
split "test Writes to Source.vi" into multiple VIs to reduce the size of the block diagram

jag2k2 commented:
> * [x]  `Creation\Creation.ctl`, `Creation\setUp.vi`, `Creation\tearDown.vi`: Remove "None" parameter from class. It is created by default and is not part of your tests.

Fixed in commit: [73dc3e0](https://github.com/ni/labview-memory-management-tools/commit/73dc3e04177477f4d3fb96bd885fec47a754e133)

> * [x]  Remove `temp_VI_UnderTest.vi` from test classes. This is an example that is not needed in your test class. You will have to remove the reference in `testExample.vit`.

Fixed in commit: [438dc21](https://github.com/ni/labview-memory-management-tools/commit/438dc21d805057ab70d6d2819b81fd83bd725b9f)

> * [x]  `Creation\testCreation256.vi`, `Creation\testCreation512.vi`, `Creation\testCreation1024.vi`: Wire error in to passIfEqual.

Fixed in commit: [55d82d4](https://github.com/ni/labview-memory-management-tools/commit/55d82d4e607d5705ebf55f6f5fa11b84cd69df86)

> * [x]  `test Writes to Source.vi`: This violates my "your block diagram should fit on a single screen" rule of thumb. Consider using subVIs or splitting into separate tests. This improves readability by allowing tech leads and developers to survey the test without scrolling 3 screens to the right.
> * [x]  `test Writes to Source.vi`: Remove stray constant on block diagram.

Fixed in commit: [6ef13a7](https://github.com/ni/labview-memory-management-tools/commit/6ef13a7f178c97c8fcd8367d27303a9b71fa0021)

> * [ ]  You have positive tests which confirm that valid parameters work as expected. Consider adding some negative tests which confirm that invalid parameters (i.e. strings that exceed static buffer length) return an error.

Hey Johann the Deterministic error builder itself doesn't throw errors but instead does what it can without affecting the size of source string.  In the case where one of these methods tries to write bytes past the size of the source string, I would expect the Replace Array Subset VIs to either crop the byte array to fit what it can into the source string or (if the offset was larger than the size of source string) have no effect at all.  Is it worth putting in some test cases to confirm these behaviors?

csjall commented:
- [ ] You still have some tests which do not wire in the error wire into the asserts: `test Status.vi`, `test Error Code.vi`, `test Source.vi`. You should always chain the incoming status.
![image](https://user-images.githubusercontent.com/4081815/63884923-cd6c0380-c99c-11e9-8779-b559ea01f75c.png)

jag2k2 committed:
chained incoming "error in" terminal for several tests

jag2k2 commented:
> * [x]  You still have some tests which do not wire in the error wire into the asserts: `test Status.vi`, `test Error Code.vi`, `test Source.vi`. You should always chain the incoming status.

Commit [e276e67](https://github.com/ni/labview-memory-management-tools/pull/8/commits/e276e67b2b898cf0dd55376ae4a4277f93c09ab5) connects the error in wire for each of the VIs you mentioned.  I went through the rest of the tests and made sure "error in" was connected.
